Src/Plugins/Security/SEC101_048.SlackWorkflowValidator.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+
+using Microsoft.CodeAnalysis.Sarif.PatternMatcher.Sdk;
+using Microsoft.RE2.Managed;
+
+namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Plugins.Security
+{
+    public class SlackWorkflowValidator : DynamicValidatorBase
+    {
+        internal const string WorkflowUri = "https://hooks.slack.com/workflows/{0:id}/{1:secret}";
+
+        protected override IEnumerable<ValidationResult> IsValidStaticHelper(IDictionary<string, FlexMatch> groups)
+        {
+            FlexMatch id = groups["id"];
+            FlexMatch secret = groups["secret"];
+
+            if (!ContainsDigitAndChar(secret.Value))
+            {
+                return ValidationResult.CreateNoMatch();
+            }
+
+            var validationResult = new ValidationResult
+            {
+                Fingerprint = new Fingerprint
+                {
+                    Id = id.Value,
+                    Secret = secret.Value,
+                    Platform = nameof(AssetPlatform.Slack),
+                },
+                ValidationState = ValidationState.Unknown,
+            };
+
+            return new[] { validationResult };
+        }
+
+        protected override ValidationState IsValidDynamicHelper(ref Fingerprint fingerprint,
+                                                                ref string message,
+                                                                IDictionary<string, string> options,
+                                                                ref ResultLevelKind resultLevelKind)
+        {
+            string id = fingerprint.Id;
+            string secret = fingerprint.Secret;
+            string uri = string.Format(WorkflowUri, id, secret);
+
+            try
+            {
+                HttpClient client = CreateOrRetrieveCachedHttpClient();
+
+                using var request = new HttpRequestMessage(HttpMethod.Post, uri);
+
+                using HttpResponseMessage response = client
+                    .SendAsync(request, HttpCompletionOption.ResponseHeadersRead)
+                    .GetAwaiter()
+                    .GetResult();
+
+                switch (response.StatusCode)
+                {
+                    case HttpStatusCode.OK:
+                    {
+                        return ValidationState.Authorized;
+                    }
+
+                    case HttpStatusCode.NotFound:
+                    {
+                        message = "The specified Slack webhook could not be found.";
```

Comment:
>webhook [](http://example.com/codeflow?start=55&length=7)

webhook? isn't that another rule?

Reply:
yeah, forgot to update.
will do once u finish this review!

thanks!